src/Spout/Reader/Helper/XLSX/SharedStringsHelper.php

```diff
@@ -34,6 +34,9 @@ class SharedStringsHelper
     /** Value to use to escape the line feed character ("\n") */
     const ESCAPED_LINE_FEED_CHARACTER = '_x000A_';
 
+    /** This will increase your memory usage but can improve your execution time */
+    static public $KEEP_ALL_STRINGS_IN_MEMORY = false;
```

Comment:
The problem here is that there is no way to know how the strings are ordered unless you actually parse the whole sheet data file. And of course, you don't want to parse the whole file twice...
So given that you can't come up with a smart way to group strings or keep some strings in memory, what compromise would you choose?

My answer to that is the following: for optimization reason, the XLSX writer in Spout does not use shared strings as they should be. A duplicate string should only have one entry in the sharedStrings.xml file and be referenced twice in the sheet data file. Instead, and because we don't want to keep in memory which strings have already been processed, every string gets assigned a new index. Not size efficient but memory efficient!
This means that shared strings are ordered by appearance, making the current caching mechanism efficient.

Because that's the only scenario that I can control, and that I can't optimize for other scenarios, I decided to optimize the caching for this one.